lib/utils/authenticatedRequest.js

```diff
@@ -132,19 +122,19 @@ class AuthenticatedRequest {
     // Only one sodium cipher is currently supported.
     const authMethod = supportedTSV1Algorithms[0]
     const nonce = uuid()
-    const headerString = `${authMethod}; ${this.publicSigningKey}; ${timestamp}; ${nonce}; uid:${userId}`
+    const headerString = `${authMethod}; ${this.config.publicSigningKey}; ${timestamp}; ${nonce}; uid:${userId}`
     const strToSign = `${path}; ${queryString}; ${options.method}; ${headerString}`
     const hashToSign = await this.crypto.genericHash(strToSign)
     const fullSignature = await this.crypto.signDetached(
       hashToSign,
-      this.privateSigningKey
+      this.config.privateSigningKey
     )
     const authHeader = `${headerString}; ${fullSignature}`
     return authHeader
   }
 
   canPerformTokenAuth() {
-    return !(this.apiKeyId === undefined) && !(this.apiSecret === undefined)
+    return this.config.apiKeyId !== undefined && this.config.apiSecret !== undefined
```

Comment:
Should this be `==` since we probably don't want `null` to be allowed?

Reply:
This applies to most of the `!== undefined` checks

Reply:
in this case I'm keeping the type check in place... mostly because I'm not sure if there are situations where a falsey value would actually be valid. Type juggling is _probably_ fine here, but I went for a 100% compatible refactor wherever possible, just cleaning some stuff up.

Reply:
Well, even within falsey values, there are classes. The confusing thing is that !(a==b) is not the same as (a!=b). But in particular, null == undefined and no other falsey value == null or == undefined